clients/stellar-relay-lib/src/config.rs

```diff
@@ -0,0 +1,277 @@
+use crate::{connection::Error, node::NodeInfo, ConnectionInfo, StellarOverlayConnection};
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, BytesOrString};
+use std::fmt::Debug;
+
+/// The configuration structure of the StellarOverlay.
+/// It configures both the ConnectionInfo and the NodeInfo.
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct StellarOverlayConfig {
+	connection_info: ConnectionInfoCfg,
+	node_info: NodeInfoCfg,
+}
+
+impl StellarOverlayConfig {
+	pub fn try_from_path(path: &str) -> Result<Self, Error> {
+		let read_file = std::fs::read_to_string(path)
+			.map_err(|e| Error::ConfigError(format!("File: {:?}", e)))?;
+		serde_json::from_str(&read_file).map_err(|e| Error::ConfigError(format!("File: {:?}", e)))
+	}
+
+	pub fn secret_key(&self) -> &[u8] {
+		&self.connection_info.secret_key
+	}
+
+	pub fn is_public_network(&self) -> bool {
+		self.node_info.is_pub_net.clone()
+	}
+
+	#[allow(dead_code)]
+	pub(crate) fn node_info(&self) -> NodeInfo {
+		self.node_info.clone().into()
+	}
+
+	#[allow(dead_code)]
+	pub(crate) fn connection_info(&self) -> Result<ConnectionInfo, Error> {
+		self.connection_info.clone().try_into()
```

Comment:
Why is this `try_into()` and the other can use `into()`?

Reply:
Aah... the `connection_info` contains the `secret_key`; and converting the String to `SecretKey` datatype returns a `Result`.

Reply:
Got it. Okay to keep it like that then